sim/battle-stream.ts

```diff
@@ -164,10 +161,7 @@ export class BattleStream extends Streams.ObjectReadWriteStream<string> {
 		this._destroy();
 	}
 	_destroy() {
-		if (this.battle) {
-			this.battle.destroy();
-		}
-		this.battle = null;
```

Comment:
Wasn't this line also removed? Am I thinking of a different stream or did this never get merged or what?

Reply:
You're thinking of c45ff0e4bcd87d0101ea97b046551767d41e1c84 on #5370 which has not been merged into HEAD yet. I just redid those changes on this PR because its not like they were specific to #5370, they just were discovered there, and landing them earlier seems useful given all my PRs involving the harness seem to get bogged down :)